src/Payment.ts

```diff
@@ -3,159 +3,249 @@ import { Utils } from './Utils'
 import { User } from './User'
 import { Transaction } from './Transaction'
 import { CurrencyNetwork } from './CurrencyNetwork'
-import { PaymentOptions } from './typings'
+import {
+  TxObject,
+  TLTxObject,
+  PathObject,
+  PathRaw,
+  PaymentOptions,
+  EventFilterOptions,
+  TLEvent
+} from './typings'
 
+/**
+ * The Payment class contains all payment related funcstion. This includes
+ * trustline transfers and normal ETH transfers.
+ */
 export class Payment {
+  private _event: Event
+  private _user: User
+  private _utils: Utils
+  private _transaction: Transaction
+  private _currencyNetwork: CurrencyNetwork
 
-  private validParameters = [ 'fromBlock', 'toBlock' ]
-
-  constructor (private event: Event,
-               private user: User,
-               private utils: Utils,
-               private transaction: Transaction,
-               private currencyNetwork: CurrencyNetwork) {
+  constructor (
+    event: Event,
+    user: User,
+    utils: Utils,
+    transaction: Transaction,
+    currencyNetwork: CurrencyNetwork
+  ) {
+    this._event = event
+    this._user = user
+    this._utils = utils
+    this._transaction = transaction
+    this._currencyNetwork = currencyNetwork
   }
 
-  public prepare (
+  /**
+   * Prepares tx object for a trustlines transfer, where loaded user is sender.
```

Comment:
tx -> ethereum transaction, to be more clear

Reply:
done